### PR TITLE
Add missing main_class arg to spark job and set default values.

### PR DESCRIPTION
**Pull request by hongye-sun**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/966)

### CODE REVIEW

File: components/gcp/dataproc/submit_hadoop_job/component.yaml [hunk 1]
```diff
@@ -27,12 +27,14 @@ inputs:
   - name: cluster_name
     description: 'Required. The cluster to run the job.'
   - name: main_jar_file_uri
+    default: ''
```
Comment (Ark-kun) [inline components/gcp/dataproc/submit_hadoop_job/component.yaml]: Is it really optional?

Comment (hongye-sun) [reply components/gcp/dataproc/submit_hadoop_job/component.yaml]: yes, user can put the jar file path in the jarFileUris and specify the mainClass to point to the main entry.

Comment (hongye-sun) [reply components/gcp/dataproc/submit_hadoop_job/component.yaml]: yes, user can put the jar file path in the jarFileUris and specify the mainClass to point to the main entry.

**Pull request merged**